### ISSUE TITLE
ConcurrentSubscribe error upon reconnect

### ISSUE THREAD

**Issue by WebSpider** [skodaconnect/myskoda#123]
myskoda version 0.8.4

Last entry in log on a normal instance running HA:
```
2024-10-22 13:05:55.199 DEBUG (MainThread) [custom_components.myskoda.coordinator] Finished fetching myskoda data in 3.261 seconds (success: True)
```

Suspend the device, and resume, and this happens:

```
2024-10-22 13:31:49.231 DEBUG (MainThread) [myskoda.mqtt] Socket to MQTT broker closed.
2024-10-22 13:31:49.231 DEBUG (MainThread) [myskoda.mqtt] Scheduling to reconnect MQTT.
2024-10-22 13:31:49.231 DEBUG (MainThread) [myskoda.mqtt] Connection to MQTT broker lost, reason 7.
2024-10-22 13:31:49.232 DEBUG (MainThread) [myskoda.mqtt] Scheduling to reconnect MQTT.
2024-10-22 13:31:49.232 DEBUG (MainThread) [myskoda.mqtt] Connecting to MQTT on mqtt.messagehub.de:8883...
2024-10-22 13:31:49.232 INFO (MainThread) [myskoda.auth.authorization] Token expired. Refreshing IDK access token
2024-10-22 13:31:49.737 DEBUG (MainThread) [myskoda.myskoda] Trace: POST https://mysmob.api.connect.skoda-auto.cz/api/v1/authenticati - response: 200 (4318 bytes) {"accessToken":"ey..4U","refreshToken":"ey..Bk","idToken":"ey..z4"}
2024-10-22 13:31:49.869 DEBUG (MainThread) [myskoda.mqtt] MQTT Connected.
2024-10-22 13:31:49.869 DEBUG (MainThread) [myskoda.mqtt] Subscribing to topic: 403cc5c0-d098-415d-8843-46ed171fd9b8/TMBJC9NY2MF017352/operation-request/air-conditioning/set-air-conditioning-at-unlock
2024-10-22 13:31:49.869 DEBUG (MainThread) [myskoda.mqtt] Subscribing to topic: 403cc5c0-d098-415d-8843-46ed171fd9b8/TMBJC9NY2MF017352/operation-request/air-conditioning/set-air-conditioning-at-unlock
  File "/home/me/.virtualenvs/homeassistant-skoda/lib/python3.12/site-packages/myskoda/mqtt.py", line 150, in connect
  File "/home/me/.virtualenvs/homeassistant-skoda/lib/python3.12/site-packages/myskoda/mqtt.py", line 260, in _subscribe_to_topic
myskoda.mqtt.ConcurrentSubscribeError
```

Datacollection succeeds, and everything is fine inside HomeAssistant, but we might want to catch this

**Comment by WebSpider** [comment]
This also occurs when the host HA is running on has a networking change

**Comment by WebSpider** [comment]
I've collected some info in my dev instance (2024.10.3) with 1.6.3 while the car was idle in the parking lot:

9:13 - Boot HA, authenticate MySkoda and connect to MQTT
9:43 - Scheduled poll, success
10:13 - Connection to MQTT lost. Refresh token and reconnect to MQTT
During reconnect the following occured:
```
2024-10-29 10:13:39.116 ERROR (MainThread) [paho.mqtt.client] Caught exception in on_socket_register_write: [Errno 2] No such file or directory
```
10:13 - Scheduled poll, success, finished 10:14

Then the :hankey: begins:

```
2024-10-29 10:14:08.411 DEBUG (MainThread) [custom_components.myskoda.coordinator] Finished fetching myskoda data in 11.375 seconds (success: True)
2024-10-29 10:14:47.456 ERROR (MainThread) [paho.mqtt.client] Caught exception in on_socket_register_write: [Errno 2] No such file or directory
2024-10-29 10:18:44.692 ERROR (MainThread) [paho.mqtt.client] Caught exception in on_socket_register_write: [Errno 2] No such file or directory
2024-10-29 10:22:59.583 DEBUG (MainThread) [myskoda.mqtt] Socket to MQTT broker closed.
2024-10-29 10:22:59.583 DEBUG (MainThread) [myskoda.mqtt] Connection to MQTT broker lost, reason 7.
2024-10-29 10:22:59.584 DEBUG (MainThread) [myskoda.mqtt] Scheduling to reconnect MQTT.
2024-10-29 10:22:59.584 DEBUG (MainThread) [myskoda.mqtt] Connecting to MQTT on mqtt.messagehub.de:8883...
2024-10-29 10:22:59.686 ERROR (MainThread) [paho.mqtt.client] Caught exception in on_socket_register_write: [Errno 2] No such file or directory
2024-10-29 10:22:59.686 ERROR (MainThread) [myskoda.mqtt] Failed to connect to MQTT.
2024-10-29 10:23:02.158 ERROR (MainThread) [paho.mqtt.client] Caught exception in on_socket_register_write: [Errno 2] No such file or directory
2024-10-29 10:23:03.884 ERROR (MainThread) [paho.mqtt.client] Caught exception in on_socket_register_write: [Errno 2] No such file or directory
2024-10-29 10:23:04.688 DEBUG (MainThread) [myskoda.mqtt] Connecting to MQTT on mqtt.messagehub.de:8883...
2024-10-29 10:23:04.791 DEBUG (MainThread) [myskoda.mqtt] MQTT Connected.
2024-10-29 10:23:04.791 DEBUG (MainThread) [myskoda.mqtt] Subscribing to topic: <uid>/<vin>/operation-request/air-conditioning/set-air-conditioning-at-unlock
```
Followed by
```
2024-10-29 10:23:35.780 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved (None)
Traceback (most recent call last):
  File "/home/me/.virtualenvs/homeassistant-skoda/lib/python3.12/site-packages/asyncio_paho/client.py", line 348, in _loop_misc
    self.reconnect()
  File "/home/me/.virtualenvs/homeassistant-skoda/lib/python3.12/site-packages/paho/mqtt/client.py", line 1088, in reconnect
    return self._send_connect(self._keepalive)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/me/.virtualenvs/homeassistant-skoda/lib/python3.12/site-packages/paho/mqtt/client.py", line 2819, in _send_connect
    return self._packet_queue(command, packet, 0, 0)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/me/.virtualenvs/homeassistant-skoda/lib/python3.12/site-packages/paho/mqtt/client.py", line 3018, in _packet_queue
    self._call_socket_register_write()
  File "/home/me/.virtualenvs/homeassistant-skoda/lib/python3.12/site-packages/paho/mqtt/client.py", line 2247, in _call_socket_register_write
    on_socket_register_write(
  File "/home/me/.virtualenvs/homeassistant-skoda/lib/python3.12/site-packages/asyncio_paho/client.py", line 318, in _on_socket_register_write_asyncio
    self._event_loop.add_writer(sock, client.loop_write)
  File "/usr/lib/python3.12/asyncio/selector_events.py", line 367, in add_writer
    self._add_writer(fd, callback, *args)
  File "/usr/lib/python3.12/asyncio/selector_events.py", line 325, in _add_writer
    self._selector.modify(fd, mask | selectors.EVENT_WRITE,
  File "/usr/lib/python3.12/selectors.py", line 389, in modify
    self._selector.modify(key.fd, selector_events)
FileNotFoundError: [Errno 2] No such file or directory
2024-10-29 10:27:23.138 ERROR (MainThread) [paho.mqtt.client] Caught exception in on_socket_register_write: [Errno 2] No such file or directory
2024-10-29 10:31:42.337 ERROR (MainThread) [paho.mqtt.client] Caught exception in on_socket_register_write: [Errno 2] No such file or directory
2024-10-29 10:34:57.796 INFO (SyncWorker_6) [homeassistant.loader] Loaded ipp from homeassistant.components.ipp
2024-10-29 10:38:03.645 ERROR (MainThread) [paho.mqtt.client] Caught exception in on_socket_register_write: [Errno 2] No such file or directory
2024-10-29 10:42:22.897 ERROR (MainThread) [paho.mqtt.client] Caught exception in on_socket_register_write: [Errno 2] No such file or directory
```

But then:

10:44 - Scheduled poll, success

Followed by:

```
2024-10-29 10:44:12.201 DEBUG (MainThread) [custom_components.myskoda.coordinator] Finished fetching myskoda data in 4.165 seconds (success: True)
2024-10-29 10:48:44.235 ERROR (MainThread) [paho.mqtt.client] Caught exception in on_socket_register_write: [Errno 2] No such file or directory
2024-10-29 10:53:03.410 ERROR (MainThread) [paho.mqtt.client] Caught exception in on_socket_register_write: [Errno 2] No such file or directory
2024-10-29 10:59:24.687 ERROR (MainThread) [paho.mqtt.client] Caught exception in on_socket_register_write: [Errno 2] No such file or directory
```

I'm going to put `paho` in debugging and see what I can find out more

**Comment by dvx76** [comment]
Should be fixed by #140 🙏

**Comment by WebSpider** [comment]
Should be :pray:
